flowforge/input/SolidComponents.py

```diff
@@ -0,0 +1,484 @@
+This can be used in a factory to build each component in a system.
+"""
+solid_component_list = {}
+
+
+class Component:
+    """
+    Abstract base class for solid components in the system
+
+    Parameters
+    ----------
+    height : float
+        Height of the component
+    n_cells : int
+        Number of discretizations made for the component
+    material : str
+        Name of the material used for the component
+    azimuthal_angle : float
+        Orientation angle of the component in the azimuthal direction
+    zenith_angle : float
+        Orientation angle of the component in the zenith (polar) direction
+    cross_section : str
+        Type of cross-section used for the component
+
+    Attributes
+    ----------
+    volume : float
+        Volume of the component
+    height : float
+        Height of the component
+    nCells : int
+        Number of discretizations made for the component
+    material : str
+        Name of the material used for the component
+    azimuthalAngle : float
+        Orientation angle of the component in the azimuthal direction
+    zenithAngle : float
+        Orientation angle of the component in the zenith (polar) direction
+    crossSection : SolidCrossSections.CrossSection
+        Cross section used for the component
+    """
+
+    def __init__(
+        self,
+        height: float,
+        n_cells: int,
+        material: str = "graphite",
+        azimuthal_angle: float = 0.0,
+        zenith_angle: float = 0.0,
+        cross_section: str = None,
+        **kwargs,
+    ) -> None:
+        self._height = height
+        self._n_cells = n_cells
+        self._material = material
+
+        if cross_section is not None:
+            self._cross_section = self._buildCrossSection(cross_section, **kwargs)
+        else:
+            self._cross_section = None
+
+        # TODO: add azimuthal and zenith angle functionality
+        assert azimuthal_angle == 0.0, "Cannot currently handle non-zero azimuthal angles"
+        assert zenith_angle == 0.0, "Cannot currently handle non-zero zenith angles"
+        self._azimuthal_angle = azimuthal_angle
+        self._zenith_angle = zenith_angle
+
+    @property
+    def volume(self) -> float:
+        return self.crossSection.area * self.height
+
+    @property
+    def height(self) -> float:
+        return self._height
+
+    @height.setter
+    def height(self, height: float):
+        self._height = height
+
+    @property
+    def nCells(self) -> int:
+        return self._n_cells
+
+    @nCells.setter
+    def nCells(self, n_cells: int):
+        self._n_cells = n_cells
+
+    @property
+    def material(self) -> str:
+        return self._material
+
+    @material.setter
+    def material(self, material: str):
+        self._material = material
+
+    @property
+    def azimuthalAngle(self) -> float:
+        assert self.azimuthalAngle == 0.0, "Cannot currently handle non-zero azimuthal angles"
```

Comment:
This assertion should be in a setter, not in a getter.

In any case, the way it is written currently, it will call itself in an infinite loop.  If you keep it, it should be:
  ```suggestion
        assert self._azimuthalAngle == 0.0, "Cannot currently handle non-zero azimuthal angles"
```

Reply:
Yep! I totally agree, and thanks for the catch on the infinite loop! Went ahead and changed this